static/app/views/issueList/groupStatsProvider.tsx

```diff
@@ -0,0 +1,129 @@
+import {createContext, useContext} from 'react';
+import {dropUndefinedKeys} from '@sentry/utils';
+import * as reactQuery from '@tanstack/react-query';
+
+import {ApiResult} from 'sentry/api';
+import type {Group, GroupStats, Organization, PageFilters} from 'sentry/types';
+import {getUtcDateString} from 'sentry/utils/dates';
+import {UseQueryResult} from 'sentry/utils/queryClient';
+import RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+
+function getEndpointParams(
+  p: Pick<GroupStatsProviderProps, 'selection' | 'period' | 'query' | 'groupIds'>
+): StatEndpointParams {
+  const params: StatEndpointParams = {
+    project: p.selection.projects,
+    environment: p.selection.environments,
+    groupStatsPeriod: p.period,
+    query: p.query,
+    groups: p.groupIds,
+    ...p.selection.datetime,
+  };
+
+  if (p.selection.datetime.period) {
+    delete params.period;
+    params.statsPeriod = p.selection.datetime.period;
+  }
+  if (params.end) {
+    params.end = getUtcDateString(params.end);
+  }
+  if (params.start) {
+    params.start = getUtcDateString(params.start);
+  }
+
+  return dropUndefinedKeys(params);
+}
+
+const GroupStatsContext = createContext<UseQueryResult<
+  Record<string, GroupStats>
+> | null>(null);
+
+export function useGroupStats(group: Group): GroupStats {
+  const ctx = useContext(GroupStatsContext);
+
+  if (!ctx) {
+    return group;
+  }
+
+  return ctx.data?.[group.id] ?? group;
+}
+
+interface StatEndpointParams extends Partial<PageFilters['datetime']> {
+  environment: string[];
+  groups: Group['id'][];
+  project: number[];
+  cursor?: string;
+  expand?: string | string[];
+  groupStatsPeriod?: string | null;
+  page?: number | string;
+  query?: string | undefined;
+  sort?: string;
+  statsPeriod?: string | null;
+}
+
+interface GroupStatsProviderProps {
+  children: React.ReactNode;
+  groupIds: Group['id'][];
+  organization: Organization;
+  period: string;
+
+  selection: PageFilters;
+  onRequest?: (promise: Promise<Record<string, GroupStats>>) => void;
+  query?: string;
+}
+
+export function GroupStatsProvider(props: GroupStatsProviderProps) {
```

Comment:
It might be simpler if, instead of a provider, you just exported the query hook? The requests will be deduped, so even if you put it in the group stream component only a single request would be made.

Reply:
Wouldn't that mean I would need to pass the groupIds and the params to all of the components so that they could make that same request?

Reply:
Ah right, forgot about the group IDs. Yeah, this solution makes sense